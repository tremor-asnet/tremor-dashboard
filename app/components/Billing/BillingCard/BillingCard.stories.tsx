// Libs
import type { Meta, StoryObj } from "@storybook/react";

// Components
import BillingCard from "./BillingCard";

const meta = {
  title: "Components/BillingCard",
  tags: ["autodocs"],
  component: BillingCard,
} as Meta<typeof BillingCard>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  render: () => (
    <BillingCard
      cardNumber="4562 1122 4594 7852"
      holderFullName="Jack Peterson"
      expires="11/22"
    />
  ),
};
