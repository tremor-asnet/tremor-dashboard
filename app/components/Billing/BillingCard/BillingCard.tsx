"use client";

// Libs
import { Flex, Text, Title } from "@tremor/react";
import Image from "next/image";

//Icons
import { MdWifi } from "react-icons/md";

// Constants
import { METADATA } from "@/constants";

//Styles
import "@/styles/billing.css";

const BillingCard = ({
  cardDigit,
  cardHolder,
  expires,
}: {
  cardDigit: string;
  cardHolder: string;
  expires: string;
}) => {
  return (
    <Flex
      alignItems="center"
      className="relative text-white bg-gradient-primary dark:bg-gradient-pickled p-0 rounded-lg">
      <div className="w-full h-full absolute top-0 left-0 opacity-20 bg-cover bg-billing-card"></div>
      <Flex
        flexDirection="col"
        justifyContent="start"
        alignItems="center"
        className="p-4">
        <Flex>
          <MdWifi className="text-2xl m-2 mb-8" />
        </Flex>
        <Flex className="text-xl text-white font-bold mb-12">{cardDigit}</Flex>
        <Flex>
          <Flex>
            <Flex flexDirection="col" alignItems="start">
              <Text className="text-whtie opacity-80 tracking-[0.4px]">
                Card Holder
              </Text>
              <Title className="text-white font-semibold tracking-[0.12px]">
                {cardHolder}
              </Title>
            </Flex>
            <Flex flexDirection="col" alignItems="start">
              <Text className="text-whtie opacity-80 tracking-[0.4px]">
                Expires
              </Text>
              <Title className="text-white font-semibold tracking-[0.12px]">
                {expires}
              </Title>
            </Flex>
          </Flex>
          <Flex justifyContent="end">
            <Image
              src={METADATA.MASTERCARD}
              alt="master card icon"
              width="55"
              height="41"
            />
          </Flex>
        </Flex>
      </Flex>
    </Flex>
  );
};

export default BillingCard;
