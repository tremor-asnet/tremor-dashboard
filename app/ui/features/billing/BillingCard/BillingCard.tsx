"use client";

import { useState } from "react";
import Image from "next/image";

// Components
import { Button, Flex, Text, Title } from "@tremor/react";

//Icons
import { MdWifi } from "react-icons/md";
import { FaRegEye } from "react-icons/fa6";
import { FaRegEyeSlash } from "react-icons/fa6";

// Constants
import { METADATA } from "@/constants";

//Styles
import "@/styles/billing.css";

const BillingCard = ({
  cardNumber,
  holderFullName,
  expire,
  cardLast4Digit,
}: {
  cardNumber: string;
  holderFullName: string;
  expire: string;
  cardLast4Digit: string;
}) => {
  const [isShowAmount, setIsShowAmount] = useState(false);

  const toggleShowHide = () => {
    setIsShowAmount(!isShowAmount);
  };

  return (
    <Flex
      alignItems="center"
      className="relative text-white bg-gradient-primary dark:bg-gradient-pickled p-0 rounded-lg min-h-[236px] shadow-[rgba(0,0,0,0.1)_0rem_1.25rem_1.5625rem_-0.3125rem,rgba(0,0,0,0.04)_0rem_0.625rem_0.625rem_-0.3125rem]">
      <div className="w-full h-full absolute top-0 left-0 opacity-20 bg-cover bg-billing-card z-0"></div>
      <Flex
        flexDirection="col"
        justifyContent="start"
        alignItems="center"
        className="p-4 z-10">
        <Flex>
          <MdWifi className="text-2xl m-2" />
        </Flex>
        <Flex className="mt-6 mb-8 gap-4">
          <Text className="text-xl text-white font-bold">
            {isShowAmount ? cardNumber : `**** **** **** ${cardLast4Digit}`}
          </Text>
          <Button
            variant="light"
            icon={isShowAmount ? FaRegEyeSlash : FaRegEye}
            onClick={toggleShowHide}
            className="text-white hover:text-white dark:text-white dark:hover:text-white"
          />
        </Flex>
        <Flex className="2xl:py-[5px]">
          <Flex>
            <Flex
              flexDirection="col"
              alignItems="start"
              className="min-w-[138px]">
              <Text className="text-whtie opacity-80 tracking-[0.4px] ">
                Card Holder
              </Text>
              <Title className="text-white font-semibold tracking-[0.12px]">
                {holderFullName}
              </Title>
            </Flex>
            <Flex flexDirection="col" alignItems="start">
              <Text className="text-whtie opacity-80 tracking-[0.4px]">
                Expires
              </Text>
              <Title className="text-white font-semibold tracking-[0.12px]">
                {expire}
              </Title>
            </Flex>
          </Flex>
          <Flex justifyContent="end">
            <Image
              src={METADATA.MASTERCARD}
              alt="master card icon"
              width="40"
              height="30"
            />
          </Flex>
        </Flex>
      </Flex>
    </Flex>
  );
};

export default BillingCard;
