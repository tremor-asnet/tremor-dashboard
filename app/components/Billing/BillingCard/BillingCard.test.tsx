import { render } from "@testing-library/react";

// Components
import BillingCard from "./BillingCard";

describe("Testing BillingCard component", () => {
  it("Should match snapshot", () => {
    const { container } = render(
      <BillingCard
        cardNumber="4562 1122 4594 7852"
        holderFullName="Jack Peterson"
        expire="11/24"
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
