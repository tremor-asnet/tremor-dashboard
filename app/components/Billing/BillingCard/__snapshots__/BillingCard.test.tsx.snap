// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing BillingCard component Should match snapshot 1`] = `
<div>
  <div
    class="tremor-Flex-root flex w-full flex-row justify-between items-center relative text-white bg-gradient-primary dark:bg-gradient-pickled p-0 rounded-lg min-h-[234px] shadow-[rgba(0,0,0,0.1)_0rem_1.25rem_1.5625rem_-0.3125rem,rgba(0,0,0,0.04)_0rem_0.625rem_0.625rem_-0.3125rem]"
  >
    <div
      class="w-full h-full absolute top-0 left-0 opacity-20 bg-cover bg-billing-card"
    />
    <div
      class="tremor-Flex-root flex w-full flex-col justify-start items-center p-4"
    >
      <div
        class="tremor-Flex-root flex w-full flex-row justify-between items-center"
      >
        <svg
          class="text-2xl m-2"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M1 9l2 2c4.97-4.97 13.03-4.97 18 0l2-2C16.93 2.93 7.08 2.93 1 9zm8 8l3 3 3-3a4.237 4.237 0 00-6 0zm-4-4l2 2a7.074 7.074 0 0110 0l2-2C15.14 9.14 8.87 9.14 5 13z"
          />
        </svg>
      </div>
      <div
        class="tremor-Flex-root flex w-full flex-row justify-between items-center text-xl text-white font-bold mt-6 mb-10"
      >
        4562 1122 4594 7852
      </div>
      <div
        class="tremor-Flex-root flex w-full flex-row justify-between items-center 2xl:py-[5px]"
      >
        <div
          class="tremor-Flex-root flex w-full flex-row justify-between items-center"
        >
          <div
            class="tremor-Flex-root flex w-full flex-col justify-between items-start min-w-[105px] mr-6"
          >
            <p
              class="text-tremor-default dark:text-dark-tremor-content text-whtie opacity-80 tracking-[0.4px]"
            >
              Card Holder
            </p>
            <p
              class="text-tremor-title dark:text-dark-tremor-content-emphasis text-white font-semibold tracking-[0.12px]"
            >
              Jack Peterson
            </p>
          </div>
          <div
            class="tremor-Flex-root flex w-full flex-col justify-between items-start"
          >
            <p
              class="text-tremor-default dark:text-dark-tremor-content text-whtie opacity-80 tracking-[0.4px]"
            >
              Expires
            </p>
            <p
              class="text-tremor-title dark:text-dark-tremor-content-emphasis text-white font-semibold tracking-[0.12px]"
            >
              11/24
            </p>
          </div>
        </div>
        <div
          class="tremor-Flex-root flex w-full flex-row justify-end items-center"
        >
          <img
            alt="master card icon"
            data-nimg="1"
            decoding="async"
            height="41"
            loading="lazy"
            src="/_next/image?url=%2Fassets%2Fimages%2Fmastercard.webp&w=128&q=75"
            srcset="/_next/image?url=%2Fassets%2Fimages%2Fmastercard.webp&w=64&q=75 1x, /_next/image?url=%2Fassets%2Fimages%2Fmastercard.webp&w=128&q=75 2x"
            style="color: transparent;"
            width="55"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
